pkg/controller/datadogagentdeployment/datadogagentdeployment_controller.go

```diff
@@ -48,12 +50,27 @@ func init() {
 // Add creates a new DatadogAgentDeployment Controller and adds it to the Manager. The Manager will set fields on the Controller
 // and Start it when the Manager is Started.
 func Add(mgr manager.Manager) error {
-	return add(mgr, newReconciler(mgr))
+	reconciler, err := newReconciler(mgr)
+	if err != nil {
+		return err
+	}
+	return add(mgr, reconciler)
 }
 
 // newReconciler returns a new reconcile.Reconciler
-func newReconciler(mgr manager.Manager) reconcile.Reconciler {
-	return &ReconcileDatadogAgentDeployment{client: mgr.GetClient(), scheme: mgr.GetScheme(), recorder: mgr.GetEventRecorderFor("DatadogAgentDeployment")}
+func newReconciler(mgr manager.Manager) (reconcile.Reconciler, error) {
+	reconciler := &ReconcileDatadogAgentDeployment{
+		client:     mgr.GetClient(),
+		scheme:     mgr.GetScheme(),
+		recorder:   mgr.GetEventRecorderFor("DatadogAgentDeployment"),
+		forwarders: datadog.NewForwardersManager(mgr.GetClient()),
+	}
+	return reconciler, mgr.Add(reconciler.forwarders)
+}
+
+type metricForwardersManager interface {
+	Register(types.NamespacedName)
+	Start(stop <-chan struct{}) error
```

Comment:
should we have Unregister() method too?

Reply:
currently we're letting each `metricsForwarder` handling this with the `ForwarderManager`.  the forwarder sends its id to the managetr when the CR is deleted, no need to have this in the reconcile loop